quickwit/quickwit-serve/src/elastic_search_api/rest_handler.rs

```diff
@@ -345,15 +346,29 @@ async fn es_compat_index_multi_search(
                 "`_msearch` request header must define at least one index".to_string(),
             )));
         }
+
+        let mut index_ids_patterns = Vec::new();
         for index in &request_header.index {
-            validate_index_id_pattern(index).map_err(|err| {
-                SearchError::InvalidArgument(format!(
-                    "request header contains an invalid index: {}",
-                    err
-                ))
-            })?;
+            match percent_decode_str(index).decode_utf8() {
+                Ok(decode_index) => {
+                    validate_index_id_pattern(decode_index.to_string().as_str()).map_err(
```

Comment:
I don't think we need to decode the index here. We are deserializing the body from the request, so everything should be fine here.
Did you see some requests failing on the multi search endpoint?

Reply:
Yes, thank you for pointing out this. I fixed it.